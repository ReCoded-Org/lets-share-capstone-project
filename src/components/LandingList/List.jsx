import React from "react";

import BlogCard from "../BlogCard/";
import PopularItemsCard from "../PopularItemsCard/";
import CategoryCard from "../CategoryCard/";

function List(props) {
    const blogs = props.blogs;
    const items = props.items;
    const categories = props.categories;

    const list = blogs ? blogs : items ? items : categories;

    return (
        <>
            <h2 className='my-10 text-center text-2xl font-bold text-primary'>
                {blogs ? "Blogs" : items ? "Items" : null}
            </h2>
            <div className=' container mx-auto mt-10  flex grid-cols-2 flex-col gap-5 px-4 text-center sm:grid md:grid-cols-4   md:text-left  xl:grid-cols-4'>
                {"" &&
                    list.map((item) =>
                        list === blogs ? (
                            <BlogCard key={item.id} item={item} />
                        ) : categories ? (
                            <CategoryCard key={item.id} item={item} />
                        ) : (
                            <PopularItemsCard key={item.id} item={item} />
                        )
                    )}
            </div>
        </>
    );
}

export default List;
