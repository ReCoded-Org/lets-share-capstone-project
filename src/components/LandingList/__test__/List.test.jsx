"use strict";

import renderer from "react-test-renderer";

import List from "../List.jsx";
const test = [1, 2, 3, 4, 5, 6];

it("renders correctly", () => {
    const tree = renderer.create(<List test={test} />).toJSON();
    expect(tree).toMatchSnapshot();
});
