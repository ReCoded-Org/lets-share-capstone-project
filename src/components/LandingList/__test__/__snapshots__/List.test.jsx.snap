// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h2
    className="my-10 text-center text-2xl font-bold text-primary"
  />,
  <div
    className=" container mx-auto mt-10 flex grid-cols-2 flex-col gap-5 px-4 text-center sm:grid md:grid-cols-4 md:text-left xl:grid-cols-4"
  />,
]
`;
